ir/src/lib.rs

```diff
@@ -15,14 +15,17 @@ pub use transition_constraints::{NodeIndex, TransitionConstraintDegree};
 mod error;
 use error::SemanticError;
 
+pub type PublicInputs = Vec<(String, usize)>;
+pub type PeriodicColumns = Vec<Vec<u64>>;
```

Comment:
Should these be `Felt`'s even at the IR level already?

We don't need to change this now, but it might be helpful in the future when may want to evaluate constraints at a random point.